apps/meteor/tests/end-to-end/api/livechat/00-rooms.ts

```diff
@@ -1111,6 +1111,66 @@ describe('LIVECHAT - rooms', function () {
 				.expect(200);
 			await deleteVisitor(visitor.token);
 		});
+
+		it('should allow visitor to download file', async () => {
+			const visitor = await createVisitor();
+			const room = await createLivechatRoom(visitor.token);
+
+			const { body } = await request
+				.post(api(`livechat/upload/${room._id}`))
+				.set('x-visitor-token', visitor.token)
+				.attach('file', fs.createReadStream(path.join(__dirname, '../../../data/livechat/sample.png')))
+				.expect('Content-Type', 'application/json')
+				.expect(200);
+
+			const {
+				files: [{ _id, name }],
+			} = body;
+			const imageUrl = `/file-upload/${_id}/${name}`;
+			await request.get(imageUrl).query({ rc_token: visitor.token, rc_room_type: 'l', rc_rid: room._id }).expect(200);
+			await deleteVisitor(visitor.token);
+			await closeOmnichannelRoom(room._id);
+		});
+
+		it('should allow visitor to download file even after room is closed', async () => {
+			const visitor = await createVisitor();
+			const room = await createLivechatRoom(visitor.token);
+			const { body } = await request
+				.post(api(`livechat/upload/${room._id}`))
+				.set('x-visitor-token', visitor.token)
+				.attach('file', fs.createReadStream(path.join(__dirname, '../../../data/livechat/sample.png')))
+				.expect('Content-Type', 'application/json')
+				.expect(200);
+
+			await closeOmnichannelRoom(room._id);
+			const {
+				files: [{ _id, name }],
+			} = body;
+			const imageUrl = `/file-upload/${_id}/${name}`;
+			await request.get(imageUrl).query({ rc_token: visitor.token, rc_room_type: 'l', rc_rid: room._id }).expect(200);
+			await deleteVisitor(visitor.token);
```

Comment:
Kinda unrelated, but can we transfer this step to an `afterEach`? I see it's being repeated on almost every test case here
Maybe we could have `visitor` as a global variable here and delete it after every step in case it's been assigned. Just so that it's easier to read all tests

Reply:
For sure we can, and also centralize the visitor/room. But since there's a ton of cases on here, that would be a nice refactor for another task